tests/test_qnode.py

```diff
@@ -1385,7 +1405,11 @@ def circuit(x):
     )
     def test_tape_shots_set_on_call(self, shots, total_shots, shot_vector):
         """test that shots are placed on the tape if they are specified during a call."""
-        dev = qml.device("default.qubit", wires=2, shots=5)
+        with pytest.warns(
+            PennyLaneDeprecationWarning,
+            match="shots on device is deprecated",
+        ):
+            dev = qml.device("default.qubit", wires=2, shots=5)
```

Comment:
I'm thinking we should port to the new system for this test.  Also, why isn't `construct_tape` with overrriden shots raising a deprecation warning?

Reply:
@albi3ro Do you mean we `set_shots` on the qnode following up? But in that way the "overriding" wanted by this series of tests wouldn't happen at all

Reply:
Also which deprecation warning are we expect from `construct_tape` with overriden shots?

Reply:
Here we are testing:
```
"""test that shots are placed on the tape if they are specified during a call."""
```
To me, I'm interpretting that:
```
tape = qml.workflow.construct_tape(qn)(0.1, 0.2, shots=shots)
```
works, but this is also deprecated behavior that should raise a deprecation warning.

Reply:
https://github.com/PennyLaneAI/pennylane/pull/7979/commits/e788496410029b636b6a14a651a545f6b1fee062